tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_example_valid_invalid.py

```diff
@@ -384,6 +391,874 @@
+                name="EOF1I0003",
+                sections=[
+                    Section.Code(
+                        code=Op.ADDRESS + Op.POP + Op.STOP,
+                        code_inputs=0,
+                        code_outputs=NON_RETURNING_SECTION,
+                        max_stack_height=1,
+                    ),
+                    Section(
+                        kind=SectionKind.TYPE,
+                        data=Bytes.fromhex("00800001"),
+                        custom_size=4,
+                    ),
+                    Section.Data("ef"),
+                ],
+                auto_sort_sections=AutoSection.ONLY_BODY,
+            ),
+            "ef000102000100030100040400010000800001305000ef",
+            # TODO rename this to MISSING TYPE SECTION HEADER
+            # Or better if the parsers will detect that it is present
+            # but understand that it is in a wrong position and report that
+            EOFException.MISSING_TYPE_HEADER,
+            id="bad_type_section_order",
+        ),
+        pytest.param(
+            # Check that EOF1 with a bad section order fails
+            Container(
+                name="EOF1I0003",
+                sections=[
+                    Section.Code(
+                        code=Op.ADDRESS + Op.POP + Op.STOP,
+                        code_inputs=0,
+                        code_outputs=NON_RETURNING_SECTION,
+                        max_stack_height=1,
+                    ),
+                    Section.Data("ef"),
+                    Section(
+                        kind=SectionKind.TYPE,
+                        data=Bytes.fromhex("00800001"),
+                        custom_size=4,
+                    ),
+                ],
+                auto_sort_sections=AutoSection.ONLY_BODY,
+            ),
+            "ef000102000100030400010100040000800001305000ef",
+            # TODO rename this to MISSING TYPE SECTION HEADER
+            # Or better if the parsers will detect that it is present
+            # but understand that it is in a wrong position and report that
+            EOFException.MISSING_TYPE_HEADER,
+            id="bad_type_section_order_2",
+        ),
+        pytest.param(
+            # Check that EOF1 with missing types section in the header fails
+            Container(
+                name="EOF1I0004",
+                sections=[
+                    Section.Code(
+                        code=Op.ADDRESS + Op.POP + Op.STOP,
+                        code_inputs=0,
+                        code_outputs=NON_RETURNING_SECTION,
+                        max_stack_height=1,
+                    ),
+                    Section.Data("ef"),
+                ],
+                auto_type_section=AutoSection.ONLY_BODY,
+            ),
+            "ef000102000100030400010000800001305000ef",
+            EOFException.MISSING_TYPE_HEADER,
+            id="bad_type_section_order_3",
+        ),
     ],
 )
 def test_example_valid_invalid(
```

Comment:
I think we should break this function up into different ones to make it easier to find implemented tests, and we can create different functions that really test similar things. E.g. we could make a `test_deprecated_opcodes` functions and parameterize it to each opcode that is deprecated in EOF (see `V1_EOF_DEPRECATED_OPCODES` in tests/prague/eip7692_eof_v1/eip3540_eof_v1/opcodes.py, which actually needs an update), and another one for header failures, etc.

Also, I think there's definitely duplicates between this file and file `tests/prague/eip7692_eof_v1/eip3540_eof_v1/container.py`, which also should be cleaned up, but perhaps in a follow up PR.